### PR TITLE
aws-c-http: add version 0.6.22 and support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **aws-c-http/***

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — DISMISSED** [review]

**Review by uilianries — COMMENTED** [review]
Looking good, only a small detail

**Comment by toge** [comment]
@uilianries 
Please tell me the points to fix.
Unfortunately I don't find these.

File: recipes/aws-c-http/all/conanfile.py [hunk 1]
```diff
@@ -1,17 +1,17 @@
+from conan import ConanFile
+from conan.tools.files import get, copy, rmdir
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
 import os
-from conans import ConanFile, CMake, tools
 
-required_conan_version = ">=1.43.0"
+required_conan_version = ">=1.52.0"
```
Comment (uilianries) [inline recipes/aws-c-http/all/conanfile.py:6]:
```suggestion
required_conan_version = ">=1.47.0"
```

**Review by uilianries — COMMENTED** [review]
Sorry, I don't know why, but comment was lost :think: Now it's there.

**Comment by toge** [comment]
@uilianries Thanks a lot!
I fixed it.

**Review by jwillikers — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**